x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/Autoscaling.java

```diff
@@ -115,14 +118,13 @@ public Collection<Object> createComponents(
         NamedWriteableRegistry namedWriteableRegistry,
         IndexNameExpressionResolver indexNameExpressionResolver,
         Supplier<RepositoriesService> repositoriesServiceSupplier,
-        Tracer tracer
+        Tracer tracer,
+        Supplier<AllocationDeciders> allocationDecidersSupplier
     ) {
         this.clusterServiceHolder.set(clusterService);
-        return List.of(
-            new AutoscalingCalculateCapacityService.Holder(this),
-            autoscalingLicenseChecker,
-            new AutoscalingNodeInfoService(clusterService, client)
-        );
+        var capacityServiceHolder = new AutoscalingCalculateCapacityService.Holder(this);
+        reservedAutoscalingPolicyAction.set(new ReservedAutoscalingPolicyAction(capacityServiceHolder, allocationDecidersSupplier));
```

Comment:
Since we now have the list of deciders directly, could we not as well just set it directly here rather than pass it around in several places? I.e., add:

```
this.allocationDeciders.set(deciders);
```

and remove it from the argument to `ReservedAutoscalingPolicyAction` as well as from being injected into the two transport services that gets it (only to pass it to the plugin).